### DISCUSSION TITLE
Authorization using firebase

### DISCUSSION THREAD

**Discussion by NanoscaleSimulations** [reflex-dev/reflex#1392] [Q&A] [answered]
Hey, Im still new to web development, but am loving reflex so far.
I am almost finished with my app, the only thing I need to do now is create a login and signup system. How would I go about and do that with firebase?
I have read many places that you need to save some sort of token in cookies, but cannot see anything about that in Reflex. Or do I have to only use pyrebase for that?

Ant help or guidance is much appreciated 😊

**Comment by cllatser** [comment]
https://youtu.be/TBDG9BiWDvM

**Reply by NanoscaleSimulations** [reply]
Hey,
I saw this, but shouldn't there be some use of tokens and cookies for authorizing? How do you avoid that some doesn't just enter the url to site and enters without login?

**Reply by ramsy0dev** [reply, marked as answer]
@NanoscaleSimulations  you can do so by creating a function in the State of your app that will run when a user visits a route in your app, this function will check if the user is signed in if not it will redirect to the Sign-up/Sign-in page. Their is an example of a login system that you can find [here](https://github.com/reflex-dev/reflex-examples/tree/main/twitter) but it's doesn't use Firebase but i think it will give you an idea on how implement it

**Reply by NanoscaleSimulations** [reply]
Aah yes, excellent idea !
But is there a way to keep the user logged in (e.g. the rest of the day), so that they can keep visiting the site and not have to log in every time?

**Reply by ramsy0dev** [reply]
@NanoscaleSimulations you can make use of auth tokens that will get generated for each user when they sign-up and save it locally and when the user visits the site the function that i mentioned before [here](https://github.com/orgs/reflex-dev/discussions/1392#discussioncomment-6599703) will check if an auth-token exists or not and  if it's valid or not.  There is a builtin function that will let you handle reading and writing to the local storage, you can find it  [here in the documentation](https://reflex.dev/docs/api-reference/browser), it is easy to use.

(You may want to create an API for that)

**Reply by NanoscaleSimulations** [reply]
Thank you, I did not know that was possible 😊 I will look into that.

**Reply by NanoscaleSimulations** [reply]
I just realised there is a set_cookie method: https://reflex.dev/docs/api-reference/special-events
I guess, that can be used too instead of local storage.

**Reply by abdullahkhalids** [reply]
Why is there no get_cookie method?

Edit: I see the `get_cookies` method is here https://reflex.dev/docs/state/utility-methods/
